### PR TITLE
Submissions and Challenges [WIP]

### PR DESCRIPTION
**Pull request by tamarindd**
Added submissions table and updated other tables accordingly.
Pls comment with any important missing fields

### CODE REVIEW

File: server/src/controller_test.py [hunk 1]
```diff
@@ -195,4 +198,3 @@ def test_ok():
 def test_error():
     assert isinstance(controller.error('oops'), dict)
     assert controller.error('oops') == {'ok': False, 'message': 'oops'}
-
```
Comment (jacobrs) [inline server/src/controller_test.py:198]: It's actually better to keep these lines at the end. It helps version control systems.

File: server/src/schema.py [hunk 2]
```diff
@@ -26,18 +37,42 @@ class User(Base):
...
-    id = Column(Integer, autoincrement=True, primary_key=True)
+    __tablename__ = 'challenges'
+
+    identifier = Column(Integer, autoincrement=True, primary_key=True)
     value = Column(Integer)
     text = Column(String)
+    submissions = relationship("Submission", back_populates="challenge")
+
+    def __repr__(self):
+        return "<Challenge(id={}, value={}, text='{}')>".format(self.identifier, self.value, self.text)
```
Comment (jacobrs) [inline server/src/schema.py]: Codecov probably wants you to cover these __repr__ functions.

File: server/src/schema.py [hunk 3]
```diff
@@ -26,18 +37,42 @@ class User(Base):
...
         return "<User(username='{}', name='{}', score={}, team='{}')>".format(self.username, self.name, self.score, self.team_name)
 
 
 class Challenge(Base):
-    __tablename__ = 'Challenges'
-    id = Column(Integer, autoincrement=True, primary_key=True)
+    __tablename__ = 'challenges'
+
+    identifier = Column(Integer, autoincrement=True, primary_key=True)
     value = Column(Integer)
```
Comment (jacobrs) [inline server/src/schema.py]: Is this the point value of the challenge? If so, that name is a little confusing, how about worth, score, or scoreWorth?

File: server/src/schema.py [hunk 4]
```diff
@@ -26,18 +37,42 @@ class User(Base):
...
+
+    def __repr__(self):
+        return "<Challenge(id={}, value={}, text='{}')>".format(self.identifier, self.value, self.text)
+
+
+class Submission(Base):
+    __tablename__ = 'submissions'
+
+    identifier = Column(Integer, autoincrement=True, primary_key=True)
+    text = Column(String)
```
Comment (jacobrs) [inline server/src/schema.py]: what would this column represent?

Comment (lancelafontaine) [reply server/src/schema.py]: If it's what I think it is, maybe `description` is a better word?

File: server/src/controller_test.py (hunk 1 above)
Comment (tamarindd) [reply server/src/controller_test.py:198]: One empty line is enough, no?

**Pull request closed without merging**